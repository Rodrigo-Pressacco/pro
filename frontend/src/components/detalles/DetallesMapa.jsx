import { useState } from 'react';
import { GoogleMap, LoadScript, Marker, InfoWindow } from '@react-google-maps/api';
import "../../styles/detalles/DetallesMapa.css";





const DetallesMapa = (props) => {

  const center = { lat: props.latitud, lng: props.longitud };
  const containerStyle = { width: '100%', height: '540px' };
  const [visiblePopUp, setVisiblePopUp] = useState(false);
  const divStyle = { background: `white`, padding: 15 }

  return (
    <div>
      {console.log()}
      <div className="div-detalles-h3">
        <h3 className="div-mapa-h3">Â¿Donde vas a estar?</h3>
      </div>
      <div className="div-nombre-mapa">
        <h4>{props.ciudad.nombre}, {props.ciudad.pais}</h4>
        <div className="div-mapa">
          <LoadScript googleMapsApiKey="AIzaSyDjmQN7EeUNEHpE3WuoXRcTbwkUj1LFYzk" loadingElement={<p>Cargando</p>}>
            <GoogleMap
              mapContainerStyle={containerStyle}
              center={center}
              zoom={15}
              onClick={() => setVisiblePopUp(false)}
            >
              { /* Child components, such as markers, info windows, etc. */}
              <Marker position={{ lat: props.latitud, lng: props.longitud }} onClick={() => setVisiblePopUp(!visiblePopUp)} />
              {visiblePopUp ?
                <InfoWindow position={{ lat: props.latitud, lng: props.longitud }} onCloseClick={() => setVisiblePopUp(!visiblePopUp)}>
                  <div style={divStyle}>
                    <h1>{props.nombre}</h1>
                    <p>{props.ciudad.nombre}</p>
                  </div>
                </InfoWindow>
                : null}
            </GoogleMap>
          </LoadScript>
        </div>
      </div>
    </div>
  )
}
export default DetallesMapa;

/* const { isLoaded } = useJsApiLoader({
  id: 'google-map-script',
  googleMapsApiKey: "AIzaSyDjmQN7EeUNEHpE3WuoXRcTbwkUj1LFYzk"
}) */