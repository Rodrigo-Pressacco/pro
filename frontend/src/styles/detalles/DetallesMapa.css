
.div-mapa-h3{
    margin-left: 35px;
    padding: 20px 0;
    font-weight: bold;
    font-size: 30px;
    line-height: 28px;
}

.div-mapa{
    display: flex;
    justify-content: center;
    padding: 20px 0;
    width: 96%;
    margin: auto;
}

.div-nombre-mapa h4{
    margin: 20px 0 0 35px;
    font-style: normal;
    font-weight: bold;
    font-size: 20px;
    line-height: 19px;
    color: #383B58;
}

@media (max-width: 800px){
}

@media (max-width: 485px){

    

}