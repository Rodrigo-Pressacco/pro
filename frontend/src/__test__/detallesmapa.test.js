import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import DetallesMapa from '../components/detalles/DetallesMapa'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<DetallesMapa />', () => {
    let mapa, ciudad
     
    beforeEach(()=> {
        ciudad = {
            id: 1,
            nombre: 'Cleveland',
            pais: 'Estados Unidos'
            
        };
    })

    beforeEach(() => {
        mapa = shallow(<DetallesMapa ciudad={ciudad} />)
    });

    it("Deberia mostrar <DetallesMapa/> correctamente", () => {
        expect(mapa).toMatchSnapshot();
    });

    it("Deberia retornar 1 etiqueta h3", () => {
        let node = mapa.find("h3")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 1 etiqueta h4", () => {
        let node = mapa.find("h4")
        expect(node.length).toEqual(1)
    });

})