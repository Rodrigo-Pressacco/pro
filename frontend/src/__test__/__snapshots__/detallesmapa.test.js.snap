// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetallesMapa /> Deberia mostrar <DetallesMapa/> correctamente 1`] = `
<div>
  <div
    className="div-detalles-h3"
  >
    <h3
      className="div-mapa-h3"
    >
      Â¿Donde vas a estar?
    </h3>
  </div>
  <div
    className="div-nombre-mapa"
  >
    <h4>
      Cleveland
      , 
      Estados Unidos
    </h4>
    <div
      className="div-mapa"
    >
      <LoadScript
        googleMapsApiKey="AIzaSyDjmQN7EeUNEHpE3WuoXRcTbwkUj1LFYzk"
        id="script-loader"
        loadingElement={
          <p>
            Cargando
          </p>
        }
        version="weekly"
      >
        <GoogleMap
          center={
            Object {
              "lat": undefined,
              "lng": undefined,
            }
          }
          mapContainerStyle={
            Object {
              "height": "540px",
              "width": "100%",
            }
          }
          onClick={[Function]}
          zoom={15}
        >
          <Marker
            onClick={[Function]}
            position={
              Object {
                "lat": undefined,
                "lng": undefined,
              }
            }
          />
        </GoogleMap>
      </LoadScript>
    </div>
  </div>
</div>
`;
